Registry Requirements Revision (#1183)
w3c/wot: registry-analysis/Readme.md

    - Instead of WG learning each new protocol and media type, it is more efficient for people with a good understanding of the protocol or media type to write a binding.
    - Engaging other communities.
- A binding SHOULD correspond to specific TD specification version(s).

>   - Reason Behind the Requirement: A binding may not fit newer or older versions of a TD specification (e.g. `readproperty` can become `readprop` or a new operations can arrive). Thus, at the time of writing a binding, it needs to be associated to one or more known TD specification versions.

Reviewer: ```suggestion
  - Reason Behind the Requirement: A binding may not fit newer or older versions of a TD specification (e.g., `readproperty` can become `readprop`, or a new operation can arrive). Thus, at the time of writing a binding, it needs to be associated with one or more known TD specification versions.
```

- The binding registry MUST be a separate document but associated with a TD version.
  - Reasons Behind the Requirement: It is easier to update in the long term.

> - Association of a binding to the TD specification (registry entry) SHOULD be confirmed by the WoT Working Group. In other words, any person cannot say that this binding can be used with the TD version X. Thus, the custodian is the WoT WG.

Reviewer: ```suggestion
- Association of a binding with the TD specification (registry entry) SHOULD be confirmed by the WoT Working Group. In other words, a person needs some permission and/or confirmation to authoritatively say that a given binding can be used with TD version X. The custodian of this registry is the WoT WG.
```

  - Reasons Behind the Requirement: WoT WG is the manager of the registry.

> - If the WoT WG does not exist anymore, the W3C Team or their delegated entity becomes the custodian.

Reviewer: ```suggestion
- If the WoT WG no longer exists, the W3C Team or their delegated entity becomes the custodian.
```

    - It breaks interoperability otherwise
    - We should avoid confusing users, i.e. which binding should I use?
    - It would complicate implementations

>   - TODO: Clarify what happens when two ecosystems like OCF and LwM2M both use CoAP binding. The initial thinking is to register them as separate entries but clarify what they use from CoAP binding. A layered registry can be thought of as language tags with `en` extending to `en-us` and `en-uk` where the tags and entries are different but the association to `en` is expressed in the id. In our case, this would be `coap-ocf` and `coap-lwm2m`. The higher-level binding SHOULD NOT override or conflict the lower-level binding and this should be verified by the custodian. E.g. `cov:method` in CoAP binding should not be turned into `cov:operation` in the higher-level binding. The namespace (prefix and its values) defined in a binding CANNOT be redefined in any other binding.

Reviewer: ```suggestion
  - TODO: Clarify what happens when two ecosystems like OCF and LwM2M both use CoAP binding. The initial thinking is to register them as separate entries but clarify what they use from CoAP binding. A layered registry can be thought of as similar to language tags with `en` extending to `en-us` and `en-uk`, where the tags and entries are different but the association with `en` is expressed in the id. In our case, this would be `coap-ocf` and `coap-lwm2m`. A higher-level binding SHOULD NOT override or conflict with a lower-level binding, and this should be verified by the custodian, e.g., `cov:method` in CoAP binding should not be turned into `cov:operation` in the higher-level binding. The namespace (prefix and its values) defined in a binding CANNOT be redefined in any other binding.
```

  - TODO: Ask existing registry "managers" about this kind of a direction.
- A binding should correspond to a TD specification version.
- Association of a binding to the TD specification should be confirmed (or allowed) by the TD task force. In other words, any person cannot say that this binding can be used with the TD version X.
- It should be possible to register, update, or remove the association of a binding to a specific TD specification outside of the REC lifecycle of the given TD specification. This means that the process should outlive the WG if the WG ever comes to stop functioning or if the WG decides to not work on a TD specification version anymore.
- There should be no two bindings for the same protocol, media type etc.
- A binding should be identifiable by the elements in a form such as `href`, `contentType`, or other terms.
  - We need to clarify whether the URI scheme and the media type MUST be registered in IANA first.
    - Provisional registration could reduce the overhead. Any new conflicting ones would bring up a discussion but it can still result in "our" provisional getting demoted.

>     - Pros: more stable. Cons: More overhead and work

Reviewer 2: Is the possibility to register something like a URI schema or a new media type within a new Binding Template already implied here? Because I think a Binding Template document could serve as a stable reference as required by IANA, so it could potentially fill this kind of gap (if that makes sense at all or shouldn't always be done by a different/the "original" SDO).

Author: My proposal going forward with this when it comes to that:
1. Try to involve the original SDO and get them to register the URI scheme
2a. If they are not responsive, let the entry submitter register the URI scheme as provisional after having a first reviewed draft. Then let them register it permanently once the registry entry is ready. Once it is in IANA, move registry entry to final status.
2b. If there are disagreements with the original SDO, keep the registry entry in a draft status.

In all cases, keep registry entry as draft until the URI scheme is registered at IANA. Ideally, I would not want to have an IANA registration requirement but it is too risky to not do that. Someone else can do that and our binding breaks... 

Case 2a can easily happen in old protocols like Modbus and 2b can happen when the original SDO had prior discussions but never resulted in a registration (MQTT has this issue I think...)
